core/deployment/src/main/java/io/quarkus/deployment/util/GenericTypeUtil.java

```diff
@@ -0,0 +1,160 @@
+package io.quarkus.deployment.util;
+
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+
+import org.jboss.jandex.ClassInfo;
+import org.jboss.jandex.ClassType;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.IndexView;
+import org.jboss.jandex.ParameterizedType;
+import org.jboss.jandex.Type;
+import org.jboss.jandex.TypeVariable;
+
+public final class GenericTypeUtil {
+
+    private static final DotName OBJECT = DotName.createSimple(Object.class.getName());
+
+    public GenericTypeUtil() {
+    }
+
+    /**
+     * Returns the captured generic types of an interface given a class that at some point in the class
+     * hierarchy implements the interface.
+     *
+     * The list contains the types in the same order as they are generic parameters defined on the interface
+     *
+     * A result is only returned iff all the generics where captured. If any of them where not defined by the class
```

Comment:
```suggestion
     * A result is only returned if all the generics were captured. If any of them where not defined by the class
```

Reply:
Actually the `iff` part was intentional - If and only if :). But I will update it to clarify